server/eclipse-project/src/main/java/us/freeandfair/corla/model/Round.java

```diff
@@ -253,45 +258,76 @@ public Integer startAuditedPrefixLength() {
   }
   
   /**
-   * @return the number of discrepancies.
+   * Ensures that a counter exists in the specified map for the specified key.
+   * 
+   * @param the_map The map.
+   * @param the_key The key.
+   */
+  private void ensureCounterExists(final Map<AuditReason, Integer> the_map,
+                                   final AuditReason the_key) {
+    if (!the_map.containsKey(the_key)) {
+      the_map.put(the_key, 0);
+    }
+  }
+  
+  /**
+   * @return the numbers of discrepancies found in the audit so far, 
+   * categorized by contest audit reason.
    */
-  public Integer discrepancies() {
-    return my_discrepancies;
+  public Map<AuditReason, Integer> discrepancies() {
+    return Collections.unmodifiableMap(my_discrepancies);
   }
   
   /**
-   * Adds a discrepancy.
+   * Adds a discrepancy for the specified audit reason. This adds it both to the 
+   * total and to the current audit round, if one is ongoing.
+   * 
+   * @param the_reason The reason.
    */
-  public void addDiscrepancy() {
-    my_discrepancies = my_discrepancies + 1;
+  public void addDiscrepancy(final AuditReason the_reason) {
+    ensureCounterExists(my_discrepancies, the_reason);
+    my_discrepancies.put(the_reason, my_discrepancies.get(the_reason) + 1); 
   }
   
   /**
-   * Removes a discrepancy.
+   * Removes a discrepancy for the specified audit reason. This removes it 
+   * both from the total and from the current audit round, if one is ongoing.
+   * 
+   * @param the_reason The reason.
    */
-  public void removeDiscrepancy() {
-    my_discrepancies = my_discrepancies - 1;
+  public void removeDiscrepancy(final AuditReason the_reason) {
+    ensureCounterExists(my_discrepancies, the_reason);
+    my_discrepancies.put(the_reason, my_discrepancies.get(the_reason) - 1);
   }
   
   /**
-   * @return the number of disagreements.
+   * @return the numbers of disagreements found in the audit so far,
+   * categorized by contest audit reason.
    */
-  public Integer disagreements() {
+  public Map<AuditReason, Integer> disagreements() {
     return my_disagreements;
   }
   
   /**
-   * Adds a disagreement.
+   * Adds a disagreement for the specified audit reason. This adds it both to the 
+   * total and to the current audit round, if one is ongoing.
+   * 
+   * @param the_reason The reason.
    */
-  public void addDisagreement() {
-    my_disagreements = my_disagreements + 1;
+  public void addDisagreement(final AuditReason the_reason) {
+    ensureCounterExists(my_disagreements, the_reason);
+    my_disagreements.put(the_reason, my_discrepancies.get(the_reason) + 1);
```

Comment:
Same concern as above.

Reply:
Actually, this comment helped me see "discrepancies" instead of "disagreements" here, egen though that's not what you pointed out... so thanks! :)